base/common/TimeZone.h

```diff
@@ -0,0 +1,603 @@
+        }
+    }
+
+
+    // Methods similar to DateLUTImpl
+    inline time_t toDate(time_t t) const
+    {
+        return callAndConvert<time_t>(&DateLUTImpl::toDate, t);
+    }
+    inline time_t toDate(GlobalDayNum d) const
+    {
+        return callAndConvert<time_t>(&DateLUTImpl::toDate, d);
+    }
+
+    inline unsigned toMonth(time_t t) const
+    {
+        return callAndConvert<unsigned>(&DateLUTImpl::toMonth, t);
+    }
+    inline unsigned toQuarter(time_t t) const
+    {
+        return callAndConvert<unsigned>(&DateLUTImpl::toQuarter, t);
+    }
+    inline Int16 toYear(time_t t) const
+    {
+        return callAndConvert<Int16>(&DateLUTImpl::toYear, t);
+    }
+    inline unsigned toDayOfWeek(time_t t) const
+    {
+        return callAndConvert<unsigned>(&DateLUTImpl::toDayOfWeek, t);
+    }
+    inline unsigned toDayOfMonth(time_t t) const
+    {
+        return callAndConvert<unsigned>(&DateLUTImpl::toDayOfMonth, t);
+    }
+    inline UInt8 daysInMonth(GlobalDayNum d) const
+    {
+        return callAndConvert<UInt8>(&DateLUTImpl::daysInMonth, d);
+    }
+    inline UInt8 daysInMonth(time_t t) const
+    {
+        return callAndConvert<UInt8>(&DateLUTImpl::daysInMonth, t);
+    }
+    inline UInt8 daysInMonth(Int16 year, UInt8 month) const
+    {
+        return getLUT(year, month, 0).daysInMonth(year, month);
+    }
+    inline time_t toDateAndShift(time_t t, Int32 days) const
+    {
+        return callAndConvert<time_t>(&DateLUTImpl::toDateAndShift, t, days);
+    }
+    inline time_t toTime(time_t t) const
+    {
+        return callAndConvert<time_t>(&DateLUTImpl::toTime, t);
+    }
+    inline unsigned toHour(time_t t) const
+    {
+        return callAndConvert<unsigned>(&DateLUTImpl::toHour, t);
+    }
+    inline unsigned toMinute(time_t t) const
+    {
+        return callAndConvert<unsigned>(&DateLUTImpl::toMinute, t);
+    }
+
+    // =============================================================================================
+    // NOTE: All those methdods do not depend on LUT, can be done locally against `default_lut`
+    // {
+    inline unsigned toSecond(time_t t) const
+    {
+        return default_lut.toSecond(t);
+    }
+    inline time_t toStartOfMinute(time_t t) const
+    {
+        return default_lut.toStartOfMinute(t);
+    }
+    inline time_t toStartOfFiveMinute(time_t t) const
+    {
+        return default_lut.toStartOfFiveMinute(t);
+    }
+    inline time_t toStartOfFifteenMinutes(time_t t) const
+    {
+        return default_lut.toStartOfFifteenMinutes(t);
+    }
+    inline time_t toStartOfTenMinutes(time_t t) const
+    {
+        return default_lut.toStartOfTenMinutes(t);
+    }
+    // }
+    // =============================================================================================
+
+    inline time_t toStartOfHour(time_t t) const
+    {
+        return callAndConvert<time_t>(&DateLUTImpl::toStartOfHour, t);
```

Comment:
I think TimeZone should have the same semantic as DateLUTImpl by default and special new iface for new semantic. To preserve function names we could create wrapper object:

```
time_zone.toStartOfHour(t); // the same as date_lut.toStartOfHour()
ExtandedTime(time_zone).toStartOfHour(t); // new semantics
```